### PR TITLE
feat(Remote): fix remote types across the codebase

### PR DESCRIPTION
**Pull request by dkremniov-bc**
## Description
`Remote` type refactoring with purpose of improving soundness of types across the codebase.

### CODE REVIEW

File: packages/blockchain-wallet-v4/src/remote/types.ts [hunk 1]
```diff
@@ -19,11 +19,8 @@ const cata = function <E, A>(
     case 'RemoteFailure': {
       return obj.Failure(this.error)
     }
-    case 'RemoteSuccess': {
-      return obj.Success(this.data)
-    }
     default:
-    // no-action
+      return obj.Success(this.data)
```
Comment (mvashchuk-bc) [inline packages/blockchain-wallet-v4/src/remote/types.ts:23]: There is `notReachable` util here: https://github.com/blockchain/blockchain-wallet-v4-frontend/pull/5531/files#diff-ac10f9c3be5e1cc5f70b9486f99b45fd3e937fdf0d7672cd74f74a1d9b627f72

maybe we should use it here as well after it's merged

File: packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export default ({ api, coreSagas, networks }) => {
       yield put(stopSubmit('exchangePasswordReset'))
       yield put(actions.auth.exchangeResetPasswordSuccess(response))
     } catch (e) {
-      yield put(actions.auth.exchangeResetPasswordFailure(e))
+      yield put(actions.auth.exchangeResetPasswordFailure('Failed to reset exchange password'))
```
[the remark this answers is not in the document]
Comment (dkremniov-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts]: Moved the hardcoded error message to component to be able to apply translations.

File: packages/blockchain-wallet-v4-frontend/src/data/components/buySell/types.ts [hunk 3]
```diff
@@ -170,23 +170,23 @@ export type BuySellState = {
...
   buyQuote: RemoteDataType<PartialClientErrorProperties, BuyQuoteStateType>
-  card: RemoteDataType<string, BSCardType>
+  card: RemoteDataType<string | number | Error, BSCardType>
   cardId?: string
   cardSuccessRate?: BSCardSuccessRateType
   cards: RemoteDataType<PartialClientErrorProperties, Array<BSCardType>>
   checkoutDotComAccountCodes?: Array<string>
   checkoutDotComApiKey?: string
-  crossBorderLimits: RemoteDataType<string, CrossBorderLimits>
+  crossBorderLimits: RemoteDataType<unknown, CrossBorderLimits>
```
[the remark this answers is not in the document]
Comment (dkremniov-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/buySell/types.ts:179]: That is how reducer looks like for this key:
```
fetchCrossBorderLimitsFailure: (state, action: PayloadAction<unknown>) => {
  state.crossBorderLimits = Remote.Failure(action.payload)
}
```

And in all the sagas, we call `fetchCrossBorderLimitsFailure` with generic error from `try..catch` statement. I didn't want to make too many assumptions, so just followed the types from reducer. We do not seem to be reliant on the type of the error in the code that consumes this slice; however, I am happy to change it if you know what is the error type we are getting for this data.

**Review by TheLeoB — APPROVED** [review]
Tested the flows I saw changes for, also did some authorization testing, looks good.

**Pull request merged**